add enrollment frontend

redshiftzero opened the pull request:
This adds an enrollment frontend:

<img width="1038" height="673" alt="Screenshot 2026-01-14 at 6 17 19 PM" src="https://github.com/user-attachments/assets/8e7bbaa8-a804-4b3c-b354-143a693a1f22" />

Closes #7

You can configure the oracle endpoints in `config.json`, or it will fetch them from the chain's query API.

I test this by:
1. Running the chain locally (CometBFT + felidae), with at least one oracle configured in the genesis file
2. Running the oracle HTTP submission server: `RUST_LOG=info cargo run --bin felidae oracle server --port 8080 --node http://localhost:26657`
3. Running the frontend (via `npm run dev`), and making sure that domains that I manually submit via the web UI are observed by the oracle (easiest way is to watch the log messages from the oracle terminal)

redshiftzero committed:
add some logging in the oracle submission API for testing

redshiftzero committed:
add initial frontend

lsd-cat commented:
It looks great, thanks for putting it together this fast! Are there cases in which the user would provide a zone that we autodetect wrong?

redshiftzero committed:
feat: integrate mozilla PSL for zone detection

redshiftzero committed:
oracle: optionally infer zone in CLI

redshiftzero committed:
oracle: disallow setting zone in HTTP API, it must be inferred

redshiftzero committed:
frontend: do not let user specify zone

redshiftzero committed:
make styles match webcat landing page

redshiftzero committed:
add required client side PoW

redshiftzero committed:
client side PoW - oracles can have different secrets now

initially the oracles all used the same secret, but i changed that
such that oracles generate and store a secret on first run

redshiftzero committed:
slow down default PoW

conorsch committed:
build: add frontend to nix env

Bolts on a reference to the new frontend in the flake nix:
basically just means we add nodejs, which brings in npm, and adds a
package spec. I'd like to specify a container image for frontend too,
but will circle back on that.

conorsch committed:
fix: use bind address in frontend app

We must ensure that the frontend can bind to non-local host, in order to
serve from application contexts such as container images.

Updates the config.json spec to take URLs to oracle "endpoints", so that
we can parse the values into proto, hostname, and ports, where
necessary.
